hail/python/hailtop/batch/resource.py

```diff
@@ -448,3 +448,6 @@ def __str__(self):
 
     def __repr__(self):
         return self._uid  # pylint: disable=no-member
+
+
+ResourceType = PythonResult | ResourceFile | ResourceGroup
```

Comment:
Still have pipes here. I didn't mean to get rid of the types altogether though, just replace the pipes with a `Union`